### ISSUE TITLE
Regression in vscode 1.70 when custom BASH prompt displays the exit code of the last command

### ISSUE THREAD

**Issue by CyrilOtheninGirard** [microsoft/vscode#158073]
Type: <b>Bug</b>

VS Code version: Code 1.70.1 (6d9b74a70ca9c7733b29f0456fd8195364076dda, 2022-08-10T06:08:33.642Z)
OS version: Windows_NT x64 10.0.19044
Remote OS version: Linux x64 4.15.0-169-generic

Proposed labels: <b>\[bug\]  \[regression\]  \[terminal-shell-integration\]  \[terminal-shell-bash\]</b>

BASH version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

Steps to Reproduce:

1. In your $HOME/.bashrc, add this to create a custom prompt showing the exit code of the last command:
```
function my_prompt {
    local last_command_exit_code=$?
    PS1="\w"
    ((last_command_exit_code == 0)) || PS1+=" err=$last_command_exit_code"
    PS1+=" $ "
}
PROMPT_COMMAND="my_prompt"
```
2. ✔️It's OK in VS Code (any version) with `"terminal.integrated.shellIntegration.enabled": false`

<img width="379" alt="vscode without shell integration" src="https://user-images.githubusercontent.com/89539021/184479031-7d7d89c7-efad-4883-8d9a-f5d6c59491f3.png">

3. ✔️It's OK in VS Code 1.69.2 with `"terminal.integrated.shellIntegration.enabled": true`

<img width="379" alt="vscode 1 69 2 with shell integration" src="https://user-images.githubusercontent.com/89539021/184479117-04665d2b-e599-471c-a9e4-bc8ee696cf5b.png">

4. ❌ It **fails** in VS Code 1.70.1 with `"terminal.integrated.shellIntegration.enabled": true` 
=> displayed exit code is **always**  `err=1` ☹️

<img width="381" alt="vscode 1 70 1 with shell integration" src="https://user-images.githubusercontent.com/89539021/184479141-8bffa120-da70-4a39-929f-0aa8be6a369d.png">

Note: ⚠️this regression does not happen (is fixed ?) in VS Code 1.71.0-insider

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 PRO 3500U w/ Radeon Vega Mobile Gfx (8 x 2096)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|13.93GB (0.76GB free)|
|Process Argv|--crash-reporter-id 8497e3ec-7e0c-4eed-a7e0-970efd42b12e|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: VM|
|OS|Linux x64 4.15.0-169-generic|
|CPUs|AMD Ryzen 5 PRO 3500U w/ Radeon Vega Mobile Gfx (4 x 2095)|
|Memory (System)|5.82GB (0.58GB free)|
|VM|100%|
</details><details><summary>Extensions (39)</summary>

Extension|Author (truncated)|Version
---|---|---
systemd-unit-file|coo|1.0.6
cpp-reference|Guy|0.2.1
vscode-todo-highlight|jgc|2.0.4
remote-ssh|ms-|0.84.0
remote-ssh-edit|ms-|0.80.0
remote-wsl|ms-|0.66.3
shellman|Rem|5.6.0
shebang-snippets|rpi|0.1.4
editorconfig-vscode-snippet|zgu|0.2.0
archive-viewer|avi|0.0.1
vscode-linux-binary-preview|bet|2.4.0
cmake-format|che|0.6.11
path-intellisense|chr|2.8.1
vscode-markdownlint|Dav|0.47.0
xml|Dot|2.5.1
cpp-reference-and-documentation|Fed|0.0.5
code-runner|for|0.11.8
shell-format|fox|7.2.2
gc-excelviewer|Gra|4.2.55
path-autocomplete|ion|1.20.2
vscode-graphviz|joa|0.0.6
svn-scm|joh|2.15.5
vscode-sshfs|Kel|1.25.0
vscode-clangd|llv|0.1.21
bash-ide-vscode|mad|1.14.0
direnv|mkh|0.6.1
python|ms-|2022.12.0
vscode-pylance|ms-|2022.8.20
jupyter|ms-|2022.7.1102252217
hexeditor|ms-|1.9.8
references-view|ms-|0.0.89
bash-extension-pack|pin|2.0.0
bash-debug|rog|0.3.9
shebang-snippets|rpi|0.1.4
vscode-taskexplorer|spm|2.9.1
shellcheck|tim|0.19.6
graphviz-interactive-preview|tin|0.3.1
tldraw-vscode|tld|1.19.0
cmake|twx|0.0.17

</details>

**Comment by CyrilOtheninGirard** [comment]
Looks like the regression has been fixed in VS Code 1.71.0-insider by this change in file `shellIntegration-bash.sh` :

```
 $ diff -w ~/.vscode-server{/bin/6d9b74a70ca9c7733b29f0456fd8195364076dda,-insiders/bin/eebdf8174b087979ae6af103f6f2e2f6f9062056}/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
39a40,44
> # Send the IsWindows property if the environment looks like Windows
> if [[ "$(uname -s)" =~ ^CYGWIN*|MINGW*|MSYS* ]]; then
>       builtin printf "\x1b]633;P;IsWindows=True\x07"
> fi
>
62,63c67
<       # Send command line, escaping printf format chars %
<       builtin printf "\033]633;E;$(echo $__vsc_current_command | sed s/%/%%/g)\007"
---
>       builtin printf "\033]633;E;%s\007" "$__vsc_current_command"
157a162,165
> __vsc_restore_exit_code() {
>       return $1
> }
>
167a176
>               __vsc_restore_exit_code "${__vsc_status}"
```

**Comment by Tyriar** [comment]
Should be fixed with https://github.com/microsoft/vscode/pull/158091
